source/developers-guide/plugin-quick-start/index.md

```diff
@@ -117,195 +141,119 @@ In many cases you might want to modify the template. Shopware makes use of the S
-     *
-     * @param string $oldVersion
-     * @return bool|void
-     * @throws Exception
-     */
-    public function update($oldVersion)
-    {
-        ...
+<?php
+namespace SwagSloganOfTheDay\Subscriber;
 
-        // 5.2+ specific: manually delete old files, which are not included in this version anymore.
-        if (version_compare($oldVersion, '3.2.0', '<')) {
-            $this->unlinkOldFiles($oldVersion);
-        }
+use Enlight\Event\SubscriberInterface;
+use Shopware\Components\Plugin\ConfigReader;
+use SwagSloganOfTheDay\Components\SloganPrinter;
 
-       ...
-    }
-
-    /**
-     * Unlinks old files which are not available for this version anymore.
-     * @param string $oldVersion
-     */
-    private function unlinkOldFiles($oldVersion)
-    {
-        $filesToDelete = [];
-
-        //Initial version for the 5.2 Plugin version.
-        if (version_compare($oldVersion, '3.2.0', '<')) {
-            $filesToDelete[] = __DIR__ . '/Subscriber/ControllerPath.php';
-            $filesToDelete[] = __DIR__ . '/Views/responsive';
-            $filesToDelete[] = __DIR__ . '/Views/emotion';
-        }
-
-        //Iterate through all provided file paths and unlink every file.
-        foreach ($filesToDelete as $filePath) {
-            if (!file_exists($filePath)) {
-                continue;
-            }
-
-            //Delete file from filesystem.
-            unlink($filePath);
-        }
-    }
-```
+class RouteSubscriber implements SubscriberInterface
+{
+   private $pluginDirectory;
+   private $sloganPrinter;
+   private $config;
+
+   public static function getSubscribedEvents()
+   {
+       return [
+           'Enlight_Controller_Action_PostDispatchSecure_Frontend' => 'onPostDispatch'
+       ];
+   }
 
-### Event callback
+   public function __construct($pluginName, $pluginDirectory, SloganPrinter $sloganPrinter, ConfigReader $configReader)
+   {
+       $this->pluginDirectory = $pluginDirectory;
+       $this->sloganPrinter = $sloganPrinter;
 
-Now the event callback function should be implemented:
+       $this->config = $configReader->getByPluginName($pluginName);
+   }
 
-```
-public function onFrontendPostDispatch(Enlight_Event_EventArgs $args)
-{
-    /** @var \Enlight_Controller_Action $controller */
-    $controller = $args->get('subject');
-    $view = $controller->View();
+   public function onPostDispatch(\Enlight_Controller_ActionEventArgs $args)
+   {
+       /** @var \Enlight_Controller_Action $controller */
+       $controller = $args->get('subject');
+       $view = $controller->View();
 
-    $view->addTemplateDir(
-        __DIR__ . '/Views'
-    );
+       $view->addTemplateDir($this->pluginDirectory . '/Resources/views');
 
-    $view->assign('slogan', $this->getSlogan());
-    $view->assign('sloganSize', $this->Config()->get('font-size'));
-    $view->assign('italic', $this->Config()->get('italic'));
+       $view->assign('fontSize', $this->config['fontSize']);
```

Comment:
@philipgatzka changed theses variables in the example plugins
see here https://github.com/shopware/devdocs/pull/729

Reply:
d